### PR TITLE
Minor rec parser cleanup, build number parsing, attempt at deriving DB schema from key arrays

### PR DESCRIPTION
**Pull request by Logg-y**
Hi,

1. Rec parser cleanup: removed some unnecessary stuff, made it throw RecParserError on decompression failure instead of letting the zlib function's error propagate
2. Build number parsing: this is maybe cruder than the rest of the parser (if the format of the build string changes or they remove the ".exe" from it, this will break). I couldn't decide if exposing the whole string `AoMRT_s.exe 452295 //stream/Athens/beta` or what is presumably the build number `452295` was the best idea, given that the game is pre-release still and who knows what might change. For now I opted to just do both as I really couldn't decide on one or the other.
3. Automate the schema building from the key lists: I've tested that the resulting structures are what I think they should be, but have no way to go and test that it _actually_ works. From a code organisation perspective, I'm feel like I'm maybe stretching what should be in `/types` a bit, but I'm not sure where else this should live.

Thinking about the future where keys might get added and removed, making optional fields for the metadata interface should be pretty easy with another set of arrays, and just merging them into the final interface the same way but with Partial. I don't know anywhere near enough to have good ideas about how to handle the data already saved to the DB, or what changing the schema does to data that's already in there.

### CODE REVIEW

File: src/types/RecordedGame.ts [hunk 1]
```diff
@@ -158,4 +160,31 @@ export const MajorGodsByIndex = new Map<number, string>([
 
 export class RecParseError extends Error
 {
+}
+
+
+type MetadataSchemaEntries<T extends (...args: any)=>any> = { [key: string]: {type: T, default: ReturnType<T>, required: boolean} };
+
+function recMetadataSchemaHelper<T extends (...args: any)=>any>(keyArray: ReadonlyArray<string>, valueType: T, defaultValue: ReturnType<T>)
```
Comment (erin-fitzpatric) [inline src/types/RecordedGame.ts]: This is a clever way to build the schemas. I like it. Can you move anything db related to the src/db/mongo/schemas/recordedGame directory? Trying to keep the mongodb schemas and models seperate from the types. 

This specific helper function could move to to the utils.ts. Probably going to  rename /lib to /utils if you want to go ahead and change that before I get to it.

File: src/db/mongo/schemas/recordedGame/PlayerData.ts [hunk 2]
```diff
@@ -1,33 +1,10 @@
...
-    winratio: { type: String, default: "" },
-    ready: { type: Boolean, default: false },
-    id: { type: Number, default: 0 },
-    civishidden: { type: Boolean, default: false },
-    civwasrandom: { type: Boolean, default: false },
-    aotgblessings: { type: String, default: "" },
-    pfentity: { type: String, default: "" },
-    ugccheck: { type: Boolean, default: false }
-  },
+  RecordedGamePlayerMetadataKeySchema,
```
Comment (erin-fitzpatric) [inline src/db/mongo/schemas/recordedGame/PlayerData.ts]: I think you could go ahead and build the RecordedGamePlayerMetadataKeySchema in here if you want.

**Review by erin-fitzpatric — CHANGES_REQUESTED** [review]
Good stuff - just move the schema definitions from /types to src/db/mongo files.

**Comment by Logg-y** [comment]
Hopefully addressed the above. I also went and did the renaming /lib to /utils.

**Pull request merged**